R/sim_gs_n.R

```diff
@@ -330,6 +367,55 @@ sim_gs_n <- function(
       ans_1sim_list <- list(ans_1sim, ans_1sim_new)
       ans_1sim <- rbindlist(ans_1sim_list, use.names = TRUE, fill = TRUE)
 
+      # Get event counts per piecewise interval
+      if (!is.null(original_design) && is_logrank){
+        # Get the study duration
+        study_duration <- max(original_design$analysis$time)
+
+        # Get the change point of the piecewise HR, ended by study duration
+        fr_chg_pt <- pmin(cumsum(original_design$fail_rate$duration), study_duration)
+        pw_event <- sapply(fr_chg_pt,
+                           function(threshold) {simu_data_cut |> dplyr::filter(tte <= threshold, event == 1) |> nrow()})
```

Comment:
I can convert the {dplyr} calls to either {data.table} or base R after this PR is merged

Reply:
Thank you! Let's do it later!